pyproject.toml

```
@@ -58,7 +58,7 @@ xdoctest = "*"
 coverage = {version = "*", extras = ["toml"]}
 pytest-timeout = "^2"
 pytest-freezer = "^0.4"
-mypy = "1.9.0"
+mypy = "^1"
```

```suggestion
mypy = "1.9.0"
```
This should really remain pinned so it matches the cli. We could include a step as part of the release process to upgrade all versions of ruff, mypy etc to the latest?

---

I'm not sure I follow, as I think we should not generally do strict pinning, especially not for dev dependencies. What are the downsides for letting mypy be upgraded on `poetry update` or `poetry lock` to the newest available, compatible release?

---

Actually maybe it is ok since the CI now uses whatever version of mypy is installed in the environment. Ignore.